code/addons/test/src/postinstall.ts

```diff
@@ -51,57 +55,158 @@ export default async function postInstall(options: PostinstallOptions) {
+
+    if (!annotationsImport) {
+      reasons.push(dedent`
+        The Vitest addon cannot yet be used with ${colors.pink.bold(info.frameworkPackageName)}
+      `);
+    }
+
+    const vitestVersion = await packageManager.getInstalledVersion('vitest');
+    if (vitestVersion && !satisfies(vitestVersion, '>=2.0.0')) {
+      reasons.push(`
+        The Vitest addon requires Vitest 2.0.0 or later.
+        Please update your ${colors.pink.bold('vitest')} dependency and try again.
+      `);
+    } else {
+      const depRanges = await packageManager.getAllDependencies();
+      const checkDependencies = [...dependencies, ...optionalDependencies];
+      const latestVersions = await packageManager.getVersions(...checkDependencies);
+      const latestPackages = checkDependencies.map(
+        (pkg, index) => [pkg, latestVersions[index].replace('^', '')] as const
+      );
+
+      let hasInconsistentPackageVersions = false;
+      for (const [pkg, latestVersion] of latestPackages) {
+        if (depRanges[pkg]) {
+          if (!dependencies.includes(pkg)) {
+            // Add found optional dependency so it will be updated to the latest version
+            dependencies.push(pkg);
+          }
+
+          if (!satisfies(latestVersion, depRanges[pkg])) {
+            hasInconsistentPackageVersions = true;
+            reasons.push(dedent`
+              The package ${colors.pink.bold(pkg)} is already installed and cannot be updated to ${colors.pink.bold(latestVersion)} because it would not satisfy "${colors.pink.bold(depRanges[pkg])}".
+            `);
+          }
+        }
+      }
+
+      if (hasInconsistentPackageVersions) {
+        reasons.push(
+          'Update your dependencies and try again, or manually install the Vitest addon.'
+        );
+      }
+    }
+
+    if (info.frameworkPackageName === '@storybook/nextjs') {
+      const nextVersion = await packageManager.getInstalledVersion('next');
+      if (!nextVersion) {
+        reasons.push(dedent`
+          It seems like you are using ${colors.pink.bold('@storybook/nextjs')} without having ${colors.pink.bold('next')} installed.
+          Please install "next" or use a different Storybook framework integration and try again.
+        `);
+      }
+    }
+
+    if (reasons.length > 0) {
+      reasons.unshift(
+        'The Test addon is incompatible with your current set up and cannot be installed:'
+      );
+      reasons.push(
+        dedent`
+          To roll back the installation, remove ${colors.pink.bold(addonName)} from the "addons" array
+          in your main Storybook config file and remove the dependency from your package.json file.
+        `
+      );
+      reasons.push(
+        dedent`
+          Please check the documentation for more information about its requirements and installation:
+          ${c.cyan`https://storybook.js.org/docs/writing-tests/test-runner-with-vitest`}
+        `
+      );
+      return reasons.map((r) => r.trim()).join('\n\n');
```

Comment:
I think we want to prefix each reason with a bullet `- ` here. From the screenshot, I did not see this as a list of reasons, but from the code I see that is the intend.

Reply:
Good one!